checks.d/mapreduce.py

```diff
@@ -0,0 +1,487 @@
+mapreduce.job.successful_reduce_attempts The number of successful reduce attempts
+mapreduce.job.new_map_attempts           The number of new map attempts
+mapreduce.job.running_map_attempts       The number of running map attempts
+mapreduce.job.failed_map_attempts        The number of failed map attempts
+mapreduce.job.killed_map_attempts        The number of killed map attempts
+mapreduce.job.successful_map_attempts    The number of successful map attempts
+
+MapReduce Job Counter Metrics
+-----------------------------
+mapreduce.job.counter.reduce_counter_value   The counter value of reduce tasks
+mapreduce.job.counter.map_counter_value      The counter value of map tasks
+mapreduce.job.counter.total_counter_value    The counter value of all tasks
+
+MapReduce Task Metrics
+----------------------
+mapreduce.job.task.progress         The progress of the task as a percent
+mapreduce.job.task.elapsed_time     The elapsed time sine the application started (in ms)
+maprduce.job.task.state.new         The total number of new tasks
+maprduce.job.task.state.scheduled   The total number of tasks with a scheduled state
+maprduce.job.task.state.running     The total number of tasks with a running state
+maprduce.job.task.state.succeeded   The total number of tasks with a succeeded state
+maprduce.job.task.state.failed      The total number of tasks with a failed state
+maprduce.job.task.state.killWait    The total number of task with a kill_wait state
+maprduce.job.task.state.killed      The total number of tasks with a killed state
+maprduce.job.task.type.map          The total number of map tasks
+maprduce.job.task.type.reduce       The total number of reduce tasks
+
+MapReduce Task Counter Metrics
+------------------------------
+mapreduce.job.task.counter.map_input_records        The number of input records of a map task
+mapreduce.job.task.counter.map_output_records       The number of output records of a map task
+mapreduce.job.task.counter.combine_input_records    The number of input records of a combine task
+mapreduce.job.task.counter.combine_output_records   The number of output records of a combine task
+mapreduce.job.task.counter.reduce_input_records     The number of input records of a reduce task
+mapreduce.job.task.counter.reduce_output_records    The number of output records of a reduce task
+'''
+
+# stdlib
+from urlparse import urljoin
+from collections import defaultdict
+
+# 3rd party
+import requests
+from requests.exceptions import Timeout
+from requests.exceptions import HTTPError
+from requests.exceptions import InvalidURL
+from requests.exceptions import ConnectionError
+from simplejson import JSONDecodeError
+
+# Project
+from checks import AgentCheck
+
+# Resource Manager URI
+DEFAULT_RM_URI = 'http://localhost:8088'
+
+# Service Check Names
+SERVICE_CHECK = 'cluster.can_connect'
+YARN_SERVICE_CHECK = 'resource_manager.can_connect'
+MAPREDUCE_SERVICE_CHECK = 'application_master.can_connect'
```

Comment:
The service checks, like metrics, should be prefixed with the check name, so `mapreduce.cluster.can_connect`, etc, instead